ast/src/analyzed/mod.rs

```diff
@@ -296,8 +296,8 @@ impl<T> Analyzed<T> {
             .iter()
             .map(|identity| identity.id())
             .max()
-            .unwrap_or_default()
-            + 1;
+            .map(|id| id + 1)
+            .unwrap_or_default();
```

Comment:
this has different results tho, was it a bug?

Reply:
Maybe not a bug, but unexpected behavior: Doing [this](https://github.com/powdr-labs/powdr/blob/9eac36701e45f368094acb667d2c538017701231/ast/src/analyzed/mod.rs#L1938-L1940) would leave you with IDs 1 and 2, but I think IDs should start with zero, see [`reallocate_identity_ids`](https://github.com/powdr-labs/powdr/blob/9eac36701e45f368094acb667d2c538017701231/ast/src/analyzed/mod.rs#L337).